user_form_app/lib/app/feature/user_form/user_page_view_model.dart

```diff
@@ -1,21 +1,93 @@
 import 'package:flutter/material.dart';
+import 'package:image_picker/image_picker.dart';
 import 'package:injectable/injectable.dart';
+import 'package:my_app/app/utils/enums.dart';
 import 'package:statemanagement_riverpod/statemanagement_riverpod.dart';
 
 @injectable
 class UserPageViewModel extends BasePageViewModel {
+  //
   late TextEditingController firstNameController = TextEditingController();
   late TextEditingController lastNameController = TextEditingController();
   late TextEditingController emailNameController = TextEditingController();
   late TextEditingController phoneController = TextEditingController();
   late TextEditingController passwordController = TextEditingController();
   late TextEditingController confirmpasswordController =
       TextEditingController();
-  int _counter = 0;
-  int get counter => _counter;
 
-  void increament() {
-    _counter++;
+  /// Educational Details controllers
+  late TextEditingController gradeController = TextEditingController();
+  late TextEditingController experienceController = TextEditingController();
+  late TextEditingController designationController = TextEditingController();
+  late TextEditingController domainController = TextEditingController();
+
+  /// Address Details controllers
+
+  late TextEditingController addressController = TextEditingController();
+  late TextEditingController landMarkController = TextEditingController();
+  late TextEditingController cityController = TextEditingController();
+  late TextEditingController pinCodeController = TextEditingController();
+
+  /// Form keys to manage validation
+  final basicInfoFormKey = GlobalKey<FormState>();
+  final educationFormKey = GlobalKey<FormState>();
+  final addressFormKey = GlobalKey<FormState>();
+
+  /// To get image file
+  XFile? userImage;
+
+  /// To pick image from gallery
+  void pickImage() async {
```

Comment:
we need to create a usecase for this and also a port and adapter for image picker as it is a kind of service we use.

Reply:
so that if we need to reuse the image picker we just use the usecase we have created and dont repeat this code